linkerd/app/inbound/src/prevent_loop.rs

```diff
@@ -0,0 +1,57 @@
+use super::endpoint::{Target, TcpEndpoint};
+use linkerd2_app_core::admit;
+
+/// A connection policy that drops
+#[derive(Copy, Clone, Debug)]
+pub struct PreventLoop {
+    port: u16,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct LoopPrevented {
+    port: u16,
+}
+
+impl PreventLoop {
+    pub fn new(port: u16) -> Self {
+        Self { port }
+    }
+}
+
+impl admit::Admit<Target> for PreventLoop {
+    type Error = LoopPrevented;
+
+    fn admit(&mut self, ep: &Target) -> Result<(), Self::Error> {
+        tracing::debug!(addr = %ep.addr, self.port);
+        if ep.addr.port() == self.port {
+            return Err(LoopPrevented { port: self.port });
+        }
+
+        Ok(())
+    }
+}
+
+impl admit::Admit<TcpEndpoint> for PreventLoop {
+    type Error = LoopPrevented;
+
+    fn admit(&mut self, ep: &TcpEndpoint) -> Result<(), Self::Error> {
+        tracing::debug!(port = %ep.port, self.port);
+        if ep.port == self.port {
+            return Err(LoopPrevented { port: self.port });
+        }
+
+        Ok(())
+    }
+}
```

Comment:
nit/tioli: it seems like there _could_ be some kinda trait for "things that have a port" that we could use to generalize this? but, it's not repetitive enough for it to matter all that much, esp. since i can't immediately think of a way to generalize between the inbound and outbound logic...

Reply:
It would take just as much code in endpoints to satisfy it though ;p